src/tests/unit/tests/common/components/visualization-toggle.test.tsx

```diff
@@ -74,7 +74,7 @@ describe('VisualizationToggleTest', () => {
             className: props.className,
             onText: 'On',
             offText: 'Off',
-            ariaLabel: props.visualizationName,
+            ariaLabel: props.label ? props.label : props.visualizationName,
```

Comment:
It's usually best to avoid repeating conditional logic between a bit of code and its unit tests; it makes it easy to accidentally be "testing" behavior that isn't necessarily right despite matching the test.

This looks like it's probably an older test case from before we used snapshot testing; I'd probably lean towards removing this helper method entirely, updating the tests to use snapshots instead for pinning the rendered object under test, and make sure there are test cases that cover the different combinations of "label specified vs not" and "visualizationName specified vs not"

Reply:
Updated this in my last commit; lmk what you think of the new changes!